TESTING.md

```diff
@@ -0,0 +1,69 @@
+
+##### Maven project
+Since OpenSearch uses Gradle as the build tool, we don't recommend using Maven.
+Code coverage for a Maven project can be generated by [JaCoCo Maven plugin](https://www.eclemma.org/jacoco/trunk/doc/maven.html), and here is an [example](https://github.com/opensearch-project/security/blob/v1.0.0.0-rc1/pom.xml#L636).
+
+##### Limitation
+1. JaCoCo Gradle plugin can not collect code coverage properly for integration tests (tests that extend `OpenSearchIntegTestCase` or `OpenSearchRestTestCase`, see [here](https://github.com/opensearch-project/OpenSearch/blob/1.0/TESTING.md#base-classes-for-test-cases) to learn more). However, there is a workaround to collect coverage when the integration test is running on a local single node cluster. For your reference, [this file](https://github.com/opensearch-project/index-management/blob/opensearch-1.0.0.0-beta1/build-tools/coverage.gradle) is a custom Gradle pluign to help achieve, and it can be applied conditionally like [here](https://github.com/opensearch-project/index-management/blob/opensearch-1.0.0.0-beta1/build.gradle#L90-L93).
+2. Additional work is needed to get coverage report by JaCoCo for the test classes which use [PowerMock](https://github.com/powermock/powermock), because those classes must be instrumented before the test executes. Please see the [PowerMock wiki](https://github.com/powermock/powermock/wiki/Code-coverage-with-JaCoCo) for detail, and there is an example for using JaCoCo Maven plugin. But for JaCoCo Gradle pluign, there is no native support, see the [issue](https://github.com/gradle/gradle/issues/2429) in `gralde` repository for detail.
+3. JaCoCo Gradle plugin can not merge code coverage reports from different sub-projects out-of-box, when the code repository contains a [Gralde multi-project build](https://docs.gradle.org/6.6.1/userguide/multi_project_builds.html). If you need the aggrectiated code coverage report from multiple sub-projects, please check the [example](https://github.com/opensearch-project/alerting/blob/opensearch-1.0.0.0-beta1/build-tools/merged-coverage.gradle) of a custom Gradle plugin in OpenSearch Alerting plugin, or the [sample](https://docs.gradle.org/6.6.1/samples/sample_jvm_multi_project_with_code_coverage.html) from Gradle.
+
+#### OpenSearch Dashboards plugin
+OpenSearch Dashboards mainly uses [Jest](https://jestjs.io/) for unit testing, and plugins are recommended to utilize Jest as the unit testing framework.
+
+To generate code coverage report under Jest framework, adding the flag `--coverage` in the command to run unit tests. See [here](https://github.com/opensearch-project/security-dashboards-plugin/blob/opensearch-1.0.0.0-beta1/.github/workflows/unit-test.yml#L67) for an example in the CI workflow.
+
+### Code Coverage Reporting
+OpenSearch plugins are using [Codecov](https://about.codecov.io/) for code coverage reporting and analysis. The dashboard can be seen [here](https://app.codecov.io/gh/opensearch-project/).
```

Comment:
Can we use the same tool for reporting Dashboard plugins?

Reply:
Yeah, like security dahsboards plugin has already used Codecov to report the code coverage. The "OpenSearch plugins" in my words refers to the whole opensearch proejct. What do you think? 😂 To say "OpenSearch and OpenSearch Dashboards plugins" or "plugins of OpenSearch project" explicitly?

Reply:
Oh thats nice. May say something like OpenSearch/Dashboards or OpenSearch and OpenSearch Dashboards plugins or plugins in OpenSearch project also works. 
What ever you prefer, we just have to send a message that it works for all of them like magic :)

Reply:
Haha, good idea. Thanks. I modified "OpenSearch plugins" to "OpenSearch/Dashboards plugins" as you suggested.